core/src/main/java/org/elasticsearch/index/query/FuzzyQueryParser.java

```diff
@@ -19,59 +19,50 @@
 
 package org.elasticsearch.index.query;
 
-import org.apache.lucene.index.Term;
-import org.apache.lucene.search.FuzzyQuery;
-import org.apache.lucene.search.MultiTermQuery;
-import org.apache.lucene.search.Query;
 import org.elasticsearch.common.ParseField;
 import org.elasticsearch.common.inject.Inject;
-import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.common.unit.Fuzziness;
 import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.index.mapper.MappedFieldType;
-import org.elasticsearch.index.query.support.QueryParsers;
 
 import java.io.IOException;
 
-/**
- *
- */
-public class FuzzyQueryParser extends BaseQueryParserTemp {
+public class FuzzyQueryParser extends BaseQueryParser {
 
-    private static final Fuzziness DEFAULT_FUZZINESS = Fuzziness.AUTO;
     private static final ParseField FUZZINESS = Fuzziness.FIELD.withDeprecation("min_similarity");
 
-
     @Inject
     public FuzzyQueryParser() {
     }
 
     @Override
     public String[] names() {
-        return new String[]{FuzzyQueryBuilder.NAME};
+        return new String[]{ FuzzyQueryBuilder.NAME };
     }
 
     @Override
-    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+    public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
         XContentParser.Token token = parser.nextToken();
         if (token != XContentParser.Token.FIELD_NAME) {
             throw new QueryParsingException(parseContext, "[fuzzy] query malformed, no field");
         }
+        
         String fieldName = parser.currentName();
-
         Object value = null;
-        float boost = AbstractQueryBuilder.DEFAULT_BOOST;
-        Fuzziness fuzziness = DEFAULT_FUZZINESS;
-        int prefixLength = FuzzyQuery.defaultPrefixLength;
-        int maxExpansions = FuzzyQuery.defaultMaxExpansions;
-        boolean transpositions = FuzzyQuery.defaultTranspositions;
-        String queryName = null;
-        MultiTermQuery.RewriteMethod rewriteMethod = null;
+        
+        Fuzziness fuzziness = FuzzyQueryBuilder.DEFAULT_FUZZINESS;
+        int prefixLength = FuzzyQueryBuilder.DEFAULT_PREFIX_LENGTH;
+        int maxExpansions = FuzzyQueryBuilder.DEFAULT_MAX_EXPANSIONS;
+        boolean transpositions = FuzzyQueryBuilder.DEFAULT_TRANSPOSITIONS;
+        String rewrite = null;
         if (parseContext.isFilter()) {
-            rewriteMethod = MultiTermQuery.CONSTANT_SCORE_REWRITE;
+            rewrite = "constant_score";
```

Comment:
Sorry I might be missing something. rewrite is a String and rewriteMethod is a RewriteMethod object. So we can't assign MultiTermQuery.CONSTANT_SCORE_REWRITE to rewrite, unless we make rewrite a RewriteMethod.